src/Package/Impl/Utilities/SessionUtilities.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Host.Client;
+using Microsoft.R.Host.Client.Session;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.R.Package.Shell;
+
+namespace Microsoft.VisualStudio.R.Package.Utilities {
+    public static class SessionUtilities {
+        public static IRSession GetInteractiveSession() {
+            var workflowProvider = VsAppShell.Current.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
+            var interactiveWorkflow = workflowProvider.GetOrCreate();
+            return interactiveWorkflow.RSession;
+        }
+
+        public static async Task<string> GetRWorkingDirectoryAsync(IRInteractiveWorkflow workflow = null) {
```

Comment:
This code can be moved to the Microsoft.R.Host.Client:
```{csharp}
public static async Task<string> GetRWorkingDirectoryAsync(this IRSession session) {
    if (session.IsHostRunning) {
        await TaskUtilities.SwitchToBackgroundThread();
        try {
            using (var evaluation = await session.BeginEvaluationAsync(false)) {
               return await evaluation.GetWorkingDirectory();
            }
        } catch (OperationCanceledException) { }
    }
    return null;
}
```

Also, `IRInteractiveWorkflow` is usually assumed to be not null.